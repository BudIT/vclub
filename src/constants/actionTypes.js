export const INITIALIZE = 'initialize';
export const MEMBER_ENTER = 'member-enter';
export const MEMBER_LEAVE = 'member-leave';
