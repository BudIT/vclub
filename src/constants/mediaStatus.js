export const MediaStatusPending = 'PENDING';
export const MediaStatusReady = 'READY';
export const MediaStatusDismissed = 'DISMISSED';
export const MediaStatusDenied = 'DENIED';
export const MediaStatusNoTracks = 'NOTRACKS';
export const MediaStatusUnknown = 'UNKNOWN';
