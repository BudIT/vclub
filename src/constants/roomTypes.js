export const ChatRoomType = 'CHAT';
export const SharingRoomType = 'SHARING';
export const MediaRoomType = 'MEDIA';
export const WhiteboardRoomType = 'WHITEBOARD';
