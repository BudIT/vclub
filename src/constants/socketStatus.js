export const SocketStatusConnecting = 'CONNECTING';
export const SocketStatusConnected = 'CONNECTED';
export const SocketStatusDisconnected = 'DISCONNECTED';
