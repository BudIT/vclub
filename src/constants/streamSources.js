export const StreamSourceWebcam = 'WEBCAM';
export const StreamSourceScreen = 'SCREEN';
