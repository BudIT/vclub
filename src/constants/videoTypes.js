export const WebcamVideoType = 'WEBCAM';
export const ScreenVideoType = 'SCREEN';
