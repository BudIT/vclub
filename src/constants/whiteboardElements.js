export const RECT = 1;
export const CIRC = 2;
export const LINE = 3;
export const ELLS = 4;
export const TEXT = 5;
export const INPUT = 6;
