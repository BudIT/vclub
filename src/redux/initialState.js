export default {
  club: {
    members: [], // array of { id: String, name: String, master: Boolean }
    auth: {
      authenticated: false,
      authenticating: false,
      user: null, // { id: String, name: String, master: Boolean }
    },
  },
  site: {},
};
