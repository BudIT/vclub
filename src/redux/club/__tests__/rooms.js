/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */
import { ChatRoomType, MediaRoomType } from 'vclub/constants/roomTypes';
import initialState from 'vclub/redux/initialClubState';
import reducer, {
  CHANGE_ROOM, changeRoom,
} from '../rooms';

// action creator
test('changeRoom action creates proper action', () => {
  const nextRoom = ChatRoomType;
  const expectedAction = {
    type: CHANGE_ROOM,
    payload: nextRoom,
  };
  expect(changeRoom(nextRoom)).toEqual(expectedAction);
});

// reducer
test('reducer returns initial state', () => {
  const { rooms: roomsInitialState } = initialState;
  expect(reducer(undefined, {}))
    .toEqual(roomsInitialState);
});

test('reducer should handle CHANGE_ROOM', () => {
  const { rooms: roomsInitialState } = initialState;

  let nextRoom = ChatRoomType;
  if (roomsInitialState.currentRoom === ChatRoomType) {
    nextRoom = MediaRoomType;
  }

  const expectedState = {
    ...roomsInitialState,
    currentRoom: nextRoom,
  };

  expect(reducer(roomsInitialState, changeRoom(nextRoom))).toEqual(expectedState);
});

test('reducer should return the same state if action\'s payload already in the state', () => {
  const { rooms: roomsInitialState } = initialState;

  const currentRoom = roomsInitialState.currentRoom;
  expect(reducer(roomsInitialState, changeRoom(currentRoom))).toBe(roomsInitialState);
});
