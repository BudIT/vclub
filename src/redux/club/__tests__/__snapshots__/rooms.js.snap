exports[`test changeRoom action creates proper action 1`] = `
Object {
  "meta": Object {
    "broadcast": true,
    "remote": true,
  },
  "payload": "CHAT",
  "type": "club/rooms/change-room",
}
`;
